library/auth.inc

```diff
@@ -84,7 +84,11 @@ if (
     // Logout
     // If session has timed out / been destroyed, logout record for null user/provider will be invalid.
     if (!empty($_SESSION['authUser']) && !empty($_SESSION['authProvider'])) {
-        EventAuditLogger::instance()->newEvent("logout", $_SESSION['authUser'], $_SESSION['authProvider'], 1, "success");
+        if ((isset($_GET['timeout'])) && ($_GET['timeout'] == "1")) {
+            EventAuditLogger::instance()->newEvent("logout", $_SESSION['authUser'], $_SESSION['authProvider'], 1, "success (timeout)");
+        } else {
+            EventAuditLogger::instance()->newEvent("logout", $_SESSION['authUser'], $_SESSION['authProvider'], 1, "success");
+        }
     }
     authCloseSession();
     authLoginScreen(true);
```

Comment:
Probably not for this commit, but at some point it may be useful to move functions authCloseSession and authLoginScreen into a different module so they can be accessed directly instead of going through a dedicated iframe.